In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

In [2]:
train_df = pd.read_csv('train.csv')
train_df.tail()  # Проверка корректности импорта.

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
9995   1260          61    2.0  49.090728   33.272626            6.0      3   
9996  16265          27    2.0  64.307684   37.038420            9.0     13   
9997   2795         178    1.0  29.648057   16.555363            5.0      3   
9998  14561          21    1.0  32.330292   22.326870            5.0      3   
9999   7202          94    1.0  35.815476   22.301367            6.0      9   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
9995        12.0       1981   0.300323         B         B        52   
9996         0.0       1977   0.072158         B         B         2   
9997         5.0       1958   0.460556         B         B        20   
9998         9.0       1969   0.194489         B         B        47   
9999         9.0       1975   0.127376         B         B        43   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
9995     10311         6           NaN            1        9       B   
9996       629         1           NaN            0        0       A   
9997      4386        14           NaN            1        5       B   
9998      8004         3         125.0            3        5       B   
9999      8429         3           NaN            3        9       B   

              Price  
9995  119367.455796  
9996  199715.148807  
9997  165953.912580  
9998  171842.411855  
9999  177685.627486

In [3]:
train_df.dtypes

Id                 int64
DistrictId         int64
Rooms            float64
Square           float64
LifeSquare       float64
KitchenSquare    float64
Floor              int64
HouseFloor       float64
HouseYear          int64
Ecology_1        float64
Ecology_2         object
Ecology_3         object
Social_1           int64
Social_2           int64
Social_3           int64
Healthcare_1     float64
Helthcare_2        int64
Shops_1            int64
Shops_2           object
Price            float64
dtype: object

## Задание 1.
Посчитать основные статистики по каждому признаку (минимальные, максимальные, средние значения). Сделать выводы об адекватности этих статистик.

In [4]:
train_df.describe()

Id    DistrictId         Rooms        Square   LifeSquare  \
count  10000.00000  10000.000000  10000.000000  10000.000000  7887.000000   
mean    8383.40770     50.400800      1.890500     56.315775    37.199645   
std     4859.01902     43.587592      0.839512     21.058732    86.241209   
min        0.00000      0.000000      0.000000      1.136859     0.370619   
25%     4169.50000     20.000000      1.000000     41.774881    22.769832   
50%     8394.50000     36.000000      2.000000     52.513310    32.781260   
75%    12592.50000     75.000000      2.000000     65.900625    45.128803   
max    16798.00000    209.000000     19.000000    641.065193  7480.592129   

       KitchenSquare         Floor    HouseFloor     HouseYear     Ecology_1  \
count   10000.000000  10000.000000  10000.000000  1.000000e+04  10000.000000   
mean        6.273300      8.526700     12.609400  3.990166e+03      0.118858   
std        28.560917      5.241148      6.775974  2.005003e+05      0.119025   
min         0.000000      1.000000      0.000000  1.910000e+03      0.000000   
25%         1.000000      4.000000      9.000000  1.974000e+03      0.017647   
50%         6.000000      7.000000     13.000000  1.977000e+03      0.075424   
75%         9.000000     12.000000     17.000000  2.001000e+03      0.195781   
max      2014.000000     42.000000    117.000000  2.005201e+07      0.521867   

           Social_1      Social_2      Social_3  Healthcare_1   Helthcare_2  \
count  10000.000000  10000.000000  10000.000000   5202.000000  10000.000000   
mean      24.687000   5352.157400      8.039200   1142.904460      1.319500   
std       17.532614   4006.799803     23.831875   1021.517264      1.493601   
min        0.000000    168.000000      0.000000      0.000000      0.000000   
25%        6.000000   1564.000000      0.000000    350.000000      0.000000   
50%       25.000000   5285.000000      2.000000    900.000000      1.000000   
75%       36.000000   7227.000000      5.000000   1548.000000      2.000000   
max       74.000000  19083.000000    141.000000   4849.000000      6.000000   

            Shops_1          Price  
count  10000.000000   10000.000000  
mean       4.231300  214138.857399  
std        4.806341   92872.293865  
min        0.000000   59174.778028  
25%        1.000000  153872.633942  
50%        3.000000  192269.644879  
75%        6.000000  249135.462171  
max       23.000000  633233.466570

In [5]:
train_df.describe().loc[['min', 'max', 'mean'], :]

Id  DistrictId    Rooms      Square   LifeSquare  KitchenSquare  \
min       0.0000      0.0000   0.0000    1.136859     0.370619         0.0000   
max   16798.0000    209.0000  19.0000  641.065193  7480.592129      2014.0000   
mean   8383.4077     50.4008   1.8905   56.315775    37.199645         6.2733   

        Floor  HouseFloor     HouseYear  Ecology_1  Social_1    Social_2  \
min    1.0000      0.0000  1.910000e+03   0.000000     0.000    168.0000   
max   42.0000    117.0000  2.005201e+07   0.521867    74.000  19083.0000   
mean   8.5267     12.6094  3.990166e+03   0.118858    24.687   5352.1574   

      Social_3  Healthcare_1  Helthcare_2  Shops_1          Price  
min     0.0000       0.00000       0.0000   0.0000   59174.778028  
max   141.0000    4849.00000       6.0000  23.0000  633233.466570  
mean    8.0392    1142.90446       1.3195   4.2313  214138.857399

## Задание 2.
Посчитать для каждого признака, сколько в нем уникальных значений и как часто они встречаются. Сделать выводы по каждому признаку.

1. Rooms

In [6]:
train_df[['Rooms']].value_counts()

Rooms
2.0      3880
1.0      3705
3.0      2235
4.0       150
5.0        18
0.0         8
10.0        2
19.0        1
6.0         1
dtype: int64

Надо будет посмотреть данные для 0 и более 5 комнат

.

2. Square

In [7]:
train_df[['Square']].value_counts()

Square    
641.065193    1
44.265162     1
44.316346     1
44.308079     1
44.302140     1
             ..
61.691076     1
61.690768     1
61.673568     1
61.672748     1
1.136859      1
Length: 10000, dtype: int64

Посмотреть большие площади, max сильно отличается от mean

.

3. LifeSquare

In [8]:
train_df[['LifeSquare']].value_counts()

LifeSquare 
7480.592129    1
26.894465      1
26.914978      1
26.915615      1
26.925315      1
              ..
40.548087      1
40.551421      1
40.551864      1
40.559784      1
0.370619       1
Length: 7887, dtype: int64

Посмотреть большие жилые площади, max сильно отличается от mean

.

4. KitchenSquare

In [9]:
train_df[['KitchenSquare']].value_counts()

KitchenSquare
1.0              2460
8.0              1306
5.0              1169
10.0             1075
6.0              1038
9.0               843
0.0               697
7.0               609
12.0              249
11.0              233
13.0               67
14.0               51
4.0                39
15.0               31
3.0                22
16.0               16
20.0               14
17.0               12
19.0               11
18.0                6
2.0                 4
22.0                3
32.0                2
25.0                2
30.0                2
51.0                2
112.0               2
43.0                2
41.0                2
58.0                2
37.0                2
96.0                1
84.0                1
78.0                1
75.0                1
54.0                1
72.0                1
66.0                1
123.0               1
63.0                1
62.0                1
60.0                1
73.0                1
36.0                1
53.0              

Обратить внимание на площади кухни больше 100

.

5. Floor

In [10]:
train_df[['Floor']].value_counts()

Floor
4        949
5        876
3        872
6        802
7        759
8        643
9        604
2        588
10       553
11       457
12       438
13       398
14       362
16       290
15       287
17       256
1        235
18       185
19       121
20        68
21        59
22        54
24        53
25        34
23        25
26        18
27         5
28         3
32         2
37         1
31         1
33         1
42         1
dtype: int64

С этажами все в порядке

.

6. HouseFloor

In [11]:
train_df[['HouseFloor']].value_counts()

HouseFloor
17.0          2331
9.0           1389
12.0          1074
5.0           1031
14.0           625
16.0           576
1.0            497
25.0           374
22.0           343
0.0            269
24.0           176
8.0            141
10.0           135
4.0            134
3.0            127
15.0           123
19.0           108
20.0           105
18.0            89
7.0             59
6.0             58
2.0             48
23.0            40
21.0            33
11.0            28
13.0            27
30.0            21
40.0             7
27.0             6
32.0             4
37.0             4
28.0             3
99.0             2
48.0             2
39.0             2
47.0             1
45.0             1
44.0             1
36.0             1
38.0             1
26.0             1
31.0             1
29.0             1
117.0            1
dtype: int64

Этажность. Много 0, и 117, 99 под вопросом

.

7. HouseYear

In [12]:
train_df[['HouseYear']].value_counts()

HouseYear
1977         2652
2017          308
2016          305
2015          299
2014          210
             ... 
20052011        1
1928            1
1947            1
4968            1
1910            1
Length: 97, dtype: int64

Обратить внимание на год, если количество менее 3.

.

8. Ecology_1

In [13]:
train_df[['Ecology_1']].value_counts()

Ecology_1
0.007122     587
0.000000     537
0.243205     511
0.072158     486
0.075779     346
            ... 
0.386772       6
0.011320       3
0.394094       2
0.012522       1
0.014334       1
Length: 129, dtype: int64

Экология. Много 0

.

9. Social_1

In [14]:
train_df[['Social_1']].value_counts()

Social_1
2           688
1           654
25          557
5           528
30          465
23          434
6           377
22          367
53          355
34          311
33          252
46          245
31          243
43          238
9           213
36          209
21          207
37          205
24          205
0           201
52          201
35          200
47          177
74          176
39          159
27          151
28          144
4           141
11          138
42          129
26          127
16          123
19          118
66           99
17           96
49           95
18           92
32           87
41           83
50           79
29           70
38           64
8            62
15           51
20           50
14           46
10           24
13           22
12           18
7            15
3             9
dtype: int64

Social_1 похоже все в порядке

.

10. Social_2

In [15]:
train_df[['Social_2']].value_counts()

Social_2
264         587
1564        511
629         486
1437        346
6398        245
           ... 
568           3
1423          3
662           2
528           1
507           1
Length: 142, dtype: int64

Social_2 Посмотреть сколько значений больше 10000

max сильно выше mean

.

11. Social_3

In [16]:
train_df[['Social_3']].value_counts()

Social_3
0           2997
1           1861
3           1136
4            835
2            635
6            423
5            391
141          245
9            192
7            184
11           172
8            170
10           142
16           101
14            97
27            59
48            58
56            49
84            43
24            37
87            27
59            25
93            25
19            22
20            19
23            18
39            15
73            10
45             9
37             3
dtype: int64

Social_3 Похоже все норм

.

12. Healthcare_1

In [17]:
train_df[['Healthcare_1']].value_counts()

Healthcare_1
540.0           511
30.0            348
1046.0          245
750.0           163
229.0           148
               ... 
370.0            14
32.0             12
1815.0           10
35.0              2
0.0               1
Length: 79, dtype: int64

Healthcare_1 всего 1 ноль и много незаполненных значений

.

In [18]:
train_df[['Helthcare_2']].value_counts()

Helthcare_2
0              4214
1              2183
3              1670
2              1158
5               388
4               288
6                99
dtype: int64

Helthcare_2 все хорошо

.

In [19]:
train_df[['Shops_1']].value_counts()

Shops_1
0          2316
1          1325
5          1324
2           964
4           841
3           711
6           487
11          401
9           316
7           294
8           253
23          245
10          215
15          176
16           83
19           49
dtype: int64

Shops_1 все хорошо

.

## Задание 3
Используя выводы из двух предыдущих заданий, найдите те наблюдения, где есть пропуски, выбросы (неадекватные значения признака, которые отличаются от большинства), используя фильтрацию из pandas.

1. Надо будет посмотреть данные для 0 и более 5 комнат

In [20]:
train_df.loc[(train_df['Rooms'] > 5)]

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
2004   5927          57   10.0  59.056975   36.223072           10.0     22   
2728  14865           9   10.0  60.871266   38.420681           10.0      3   
6088   8491           1   19.0  42.006046   21.779288            7.0     17   
8794  14003          99    6.0  59.414334   38.702244            6.0      7   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
2004        22.0       2002   0.090799         B         B        74   
2728         2.0       1994   0.161532         B         B        25   
6088        17.0       2014   0.007122         B         B         1   
8794         9.0       1969   0.033494         B         B        66   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
2004     19083         2           NaN            5       15       B   
2728      5648         1          30.0            2        4       B   
6088       264         0           NaN            0        1       B   
8794     10573         1        1322.0            3        8       B   

              Price  
2004  317265.323792  
2728  172329.270863  
6088   78364.616704  
8794  229661.964416

это ошибки, 1 или 2 комнатные квартиры

.

2. Посмотреть большие площади, max сильно отличается от mean

.

In [21]:
train_df.loc[(train_df['Square'] > 200)]

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
726      28           9    2.0  604.705972         NaN            1.0     17   
1776  16568          27    4.0  200.334539  201.627361           25.0      1   
1906   5548          86    5.0  275.645284  233.949309           26.0     12   
7010   2307         102    1.0  409.425181  410.639749           10.0      4   
7644   7917          27    0.0  212.932361  211.231125            0.0      2   
9229  11602          30    2.0  641.065193  638.163193           10.0     20   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
726         18.0       1977   0.161532         B         B        25   
1776         2.0       2013   0.041116         B         B        53   
1906        37.0       2011   0.161976         B         A        31   
7010         4.0       2016   0.238617         B         B        26   
7644         3.0       2008   0.211401         B         B         9   
9229        19.0       2019   0.000078         B         B        22   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
726       5648         1          30.0            2        4       B   
1776     14892         4           NaN            1        4       B   
1906      7010         5        4508.0            3        7       B   
7010      3889         6         705.0            3        6       B   
7644      1892         0           NaN            0        1       B   
9229      6398       141        1046.0            3       23       B   

              Price  
726   187717.242538  
1776  528560.506016  
1906  455264.882666  
7010   90470.430830  
7644  302211.260887  
9229  133529.681562

из выборки адекватное значение только 1906, у остальных нестыковка между Square и LifeSquare

.

3. Посмотреть большие жилые площади, max сильно отличается от mean

In [22]:
train_df.loc[(train_df['LifeSquare'] > 200)]

Id  DistrictId  Rooms      Square   LifeSquare  KitchenSquare  Floor  \
357   14990          23    2.0   48.449873   263.542020            5.0      6   
1776  16568          27    4.0  200.334539   201.627361           25.0      1   
1906   5548          86    5.0  275.645284   233.949309           26.0     12   
5183  15886          85    3.0   78.059331   461.463614           10.0     12   
7010   2307         102    1.0  409.425181   410.639749           10.0      4   
7644   7917          27    0.0  212.932361   211.231125            0.0      2   
8000  16550          27    3.0   81.694417  7480.592129            1.0      9   
9229  11602          30    2.0  641.065193   638.163193           10.0     20   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
357          5.0       1972   0.075779         B         B         6   
1776         2.0       2013   0.041116         B         B        53   
1906        37.0       2011   0.161976         B         A        31   
5183        16.0       1998   0.037178         B         B        52   
7010         4.0       2016   0.238617         B         B        26   
7644         3.0       2008   0.211401         B         B         9   
8000        17.0       2016   0.017647         B         B         2   
9229        19.0       2019   0.000078         B         B        22   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
357       1437         3           NaN            0        2       B   
1776     14892         4           NaN            1        4       B   
1906      7010         5        4508.0            3        7       B   
5183     11217         1        2300.0            1        7       B   
7010      3889         6         705.0            3        6       B   
7644      1892         0           NaN            0        1       B   
8000       469         0           NaN            0        0       B   
9229      6398       141        1046.0            3       23       B   

              Price  
357   141780.231857  
1776  528560.506016  
1906  455264.882666  
5183  394253.299978  
7010   90470.430830  
7644  302211.260887  
8000  217357.492366  
9229  133529.681562

неадекват (и скорее всего запятую перенести влево на 1 знак) 357, 5183.

в остальных случаях LifeSquare больше Square

.

3.1 Посмотреть где LifeSquare больше Square

In [23]:
train_df.loc[(train_df['Square'] < train_df['LifeSquare'])]

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
33      847          74    2.0  73.314975   74.131261            1.0      7   
64     3992           1    3.0  87.200625   87.730225            0.0     21   
77    11771          45    1.0  38.467276   39.624493            1.0     11   
100   13343          74    2.0  79.937665   82.931595            1.0      8   
111   13875          20    1.0  28.438794   32.514021            6.0      6   
...     ...         ...    ...        ...         ...            ...    ...   
9960  16258          48    3.0  96.056784   98.152802            1.0     15   
9962  14175          14    1.0  42.029175   44.682167           10.0      4   
9979   2201           6    1.0  40.043234   41.984322            1.0      3   
9987  11665           1    2.0  62.690057   64.602939           10.0     19   
9993   9289           1    3.0  84.486078   88.931509            0.0      4   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
33           6.0       1977   0.075779         B         B         6   
64          22.0       1977   0.007122         B         B         1   
77           9.0       1977   0.195781         B         B        23   
100         21.0       1977   0.075779         B         B         6   
111         10.0       2014   0.019509         B         B        37   
...          ...        ...        ...       ...       ...       ...   
9960         1.0       2017   0.041125         B         B        46   
9962         4.0       2018   0.238617         B         B        26   
9979        17.0       2014   0.243205         B         B         5   
9987        22.0       2018   0.007122         B         B         1   
9993        22.0       2016   0.007122         B         B         1   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
33        1437         3           NaN            0        2       B   
64         264         0           NaN            0        1       B   
77        5212         6           NaN            3        2       B   
100       1437         3           NaN            0        2       B   
111       7687        11         176.0            5        5       B   
...        ...       ...           ...          ...      ...     ...   
9960      9515         5           NaN            1       10       B   
9962      3889         6         705.0            3        6       B   
9979      1564         0         540.0            0        0       B   
9987       264         0           NaN            0        1       B   
9993       264         0           NaN            0        1       B   

              Price  
33    254005.381863  
64    300180.365669  
77    247440.530601  
100   226808.747168  
111   199248.114376  
...             ...  
9960  524365.550705  
9962   87556.385617  
9979  134888.083598  
9987  203908.192698  
9993  264213.558631  

[482 rows x 20 columns]

Это какая-то жесть (482 ячейки)

.

3.2 Посмотреть если LIfeSquare равно нулю

In [24]:
train_df.loc[(train_df['LifeSquare'].isnull())]

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
0     11809          27    3.0  115.027311         NaN           10.0      4   
3      2352           1    1.0   40.409907         NaN            1.0     10   
18     4581           1    1.0   47.582480         NaN            1.0     16   
20     8768          63    1.0   50.974891         NaN            1.0      7   
24     4717          27    1.0   44.203947         NaN            0.0      2   
...     ...         ...    ...         ...         ...            ...    ...   
9975   4944           6    2.0   52.573133         NaN            1.0     18   
9978   4851           1    1.0   39.314342         NaN            1.0     13   
9981   6753          62    2.0   73.092686         NaN            1.0      4   
9983    647           1    2.0   63.324025         NaN            1.0      4   
9991  11173           6    2.0   62.648779         NaN            0.0      6   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
0           10.0       2014   0.075424         B         B        11   
3           22.0       1977   0.007122         B         B         1   
18          16.0       1977   0.007122         B         B         1   
20          16.0       2018   0.238617         B         B        26   
24           0.0       1977   0.072158         B         B         2   
...          ...        ...        ...       ...       ...       ...   
9975        17.0       1977   0.243205         B         B         5   
9978        17.0       1977   0.007122         B         B         1   
9981        16.0       2017   0.072158         B         B         2   
9983        17.0       1977   0.007122         B         B         1   
9991        17.0       1977   0.243205         B         B         5   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
0         3097         0           NaN            0        0       B   
3          264         0           NaN            0        1       B   
18         264         0           NaN            0        1       B   
20        3889         6         705.0            3        6       B   
24         629         1           NaN            0        0       A   
...        ...       ...           ...          ...      ...     ...   
9975      1564         0         540.0            0        0       B   
9978       264         0           NaN            0        1       B   
9981       629         1           NaN            0        0       A   
9983       264         0           NaN            0        1       B   
9991      1564         0         540.0            0        0       B   

              Price  
0     305018.871089  
3     168106.007630  
18    128803.593810  
20    260328.413376  
24    135466.559958  
...             ...  
9975  154366.031184  
9978  137782.941940  
9981  159077.198056  
9983  183809.327114  
9991  182625.548872  

[2113 rows x 20 columns]

Очень много значений с LifeSquare равно NaN

.

4. Обратить внимание на площади кухни больше 100

In [25]:
train_df.loc[(train_df['KitchenSquare'] > 100)]

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
286    2737          27    3.0  123.430072  125.806981          123.0      5   
5002   7441          62    3.0  114.734473  112.589083          112.0      3   
6505  14679          81    1.0   32.276663   19.278394         1970.0      6   
7733  12552          58    3.0  116.405693  113.109653          112.0      3   
9520  14656          62    1.0   47.100719   46.447960         2014.0      4   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
286         10.0       2015   0.017647         B         B         2   
5002         3.0       2015   0.072158         B         B         2   
6505         1.0       1977   0.006076         B         B        30   
7733         3.0       2016   0.437885         B         B        23   
9520         1.0       2014   0.072158         B         B         2   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
286        469         0           NaN            0        0       B   
5002       629         1           NaN            0        0       A   
6505      5285         0         645.0            6        6       B   
7733      5735         3        1084.0            0        5       B   
9520       629         1           NaN            0        0       A   

              Price  
286   234194.837047  
5002  315245.521059  
6505  105539.556275  
7733  296165.936689  
9520  108337.484207

Все 5 ячеек неадекватные относительно площади кухни и остальных площадей

.

5. С этажами было все хорошо, нужно посмотреть есть где этаж больще этажности дома

5.1 посмотреть есть где этаж больше этажности дома

In [26]:
train_df.query('(HouseFloor < Floor)')

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
17     1190           8    2.0  49.396000   28.299600            5.0      6   
19      367          45    1.0  42.367289   40.012529            1.0     16   
21     8471          20    3.0  75.611065   47.494621            8.0     11   
24     4717          27    1.0  44.203947         NaN            0.0      2   
25     4521           6    2.0  65.409965         NaN            1.0     18   
...     ...         ...    ...        ...         ...            ...    ...   
9960  16258          48    3.0  96.056784   98.152802            1.0     15   
9968   9920           1    3.0  86.487550         NaN            1.0     18   
9970  13501           6    1.0  41.310416   19.117077            9.0     18   
9975   4944           6    2.0  52.573133         NaN            1.0     18   
9996  16265          27    2.0  64.307684   37.038420            9.0     13   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
17           5.0       1962   0.458174         B         A        26   
19          14.0       2015   0.195781         B         B        23   
21           9.0       1976   0.019509         B         B        37   
24           0.0       1977   0.072158         B         B         2   
25           1.0       1977   0.243205         B         B         5   
...          ...        ...        ...       ...       ...       ...   
9960         1.0       2017   0.041125         B         B        46   
9968        17.0       1977   0.007122         B         B         1   
9970        17.0       1977   0.243205         B         B         5   
9975        17.0       1977   0.243205         B         B         5   
9996         0.0       1977   0.072158         B         B         2   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
17        4439         1        1130.0            1        5       B   
19        5212         6           NaN            3        2       B   
21        7687        11         176.0            5        5       B   
24         629         1           NaN            0        0       A   
25        1564         0         540.0            0        0       B   
...        ...       ...           ...          ...      ...     ...   
9960      9515         5           NaN            1       10       B   
9968       264         0           NaN            0        1       B   
9970      1564         0         540.0            0        0       B   
9975      1564         0         540.0            0        0       B   
9996       629         1           NaN            0        0       A   

              Price  
17    186971.926566  
19    250466.980777  
21    366916.246494  
24    135466.559958  
25    191205.272608  
...             ...  
9960  524365.550705  
9968  248248.935528  
9970  114878.830713  
9975  154366.031184  
9996  199715.148807  

[1825 rows x 20 columns]

Очень много некорректных значений (этаж больще этажности дома), нужно разбираться подробнее потом

.

6. Этажность. Много 0, и 117, 99 под вопросом

In [27]:
train_df.query('(HouseFloor > 98)')

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
746    9300          74    2.0  71.747869   74.579809            9.0      5   
1678     78          30    2.0  65.773749   66.811789            1.0      8   
4657  10806           5    1.0  51.944587   48.709601            1.0      6   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
746         99.0       1977   0.075779         B         B         6   
1678       117.0       1977   0.000078         B         B        22   
4657        99.0       1977   0.150818         B         B        16   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
746       1437         3           NaN            0        2       B   
1678      6398       141        1046.0            3       23       B   
4657      3433         4        2643.0            4        5       B   

              Price  
746   243329.912579  
1678  207248.370520  
4657  296127.115515

Скорее всего это опечатки, лишняя цифра

.

6.1 Смотрим HouseFloor = 0

In [28]:
train_df.query('(HouseFloor == 0)')

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
24     4717          27    1.0  44.203947         NaN            0.0      2   
152   12947          53    2.0  56.023342         NaN            0.0     14   
159   13146          74    1.0  46.449711         NaN            0.0      6   
186   12174          62    1.0  44.954880         NaN            0.0      8   
229   15661          45    3.0  81.094922         NaN            0.0     15   
...     ...         ...    ...        ...         ...            ...    ...   
9889    629          27    1.0  42.936724         NaN            0.0      2   
9931  15686           1    1.0  47.402412         NaN            0.0     14   
9940   4711           6    1.0  41.224118         NaN            0.0      8   
9950   5383          48    3.0  98.994909         NaN            0.0      3   
9996  16265          27    2.0  64.307684    37.03842            9.0     13   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
24           0.0       1977   0.072158         B         B         2   
152          0.0       1977   0.049637         B         B        34   
159          0.0       1977   0.075779         B         B         6   
186          0.0       1977   0.072158         B         B         2   
229          0.0       1977   0.195781         B         B        23   
...          ...        ...        ...       ...       ...       ...   
9889         0.0       1977   0.011654         B         B         4   
9931         0.0       1977   0.007122         B         B         1   
9940         0.0       1977   0.243205         B         B         5   
9950         0.0       1977   0.041125         B         B        46   
9996         0.0       1977   0.072158         B         B         2   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
24         629         1           NaN            0        0       A   
152       7759         0         229.0            1        3       B   
159       1437         3           NaN            0        2       B   
186        629         1           NaN            0        0       A   
229       5212         6           NaN            3        2       B   
...        ...       ...           ...          ...      ...     ...   
9889       915         0           NaN            0        0       B   
9931       264         0           NaN            0        1       B   
9940      1564         0         540.0            0        0       B   
9950      9515         5           NaN            1       10       B   
9996       629         1           NaN            0        0       A   

              Price  
24    135466.559958  
152   247074.564832  
159   168705.450148  
186   133773.301722  
229   442472.456032  
...             ...  
9889  129674.841139  
9931  174740.301224  
9940  125612.639104  
9950  418817.004446  
9996  199715.148807  

[269 rows x 20 columns]

269 ячеек с этажностью дома равной 0 (они также входят в 1825 ячеек где этажность дома меньше этажа)

.

7. Обратить внимание на год, если количество менее 3.

In [29]:
train_df[["HouseYear"]].value_counts()[train_df[["HouseYear"]].value_counts() < 3]

HouseYear
1941         2
1948         2
1932         2
1929         2
1912         2
1914         1
1916         1
20052011     1
1928         1
1947         1
4968         1
1910         1
dtype: int64

Неадекатные значения 4968 и 20052011

7.1 Проверить года более 2018 и менее 1900

In [30]:
train_df.query('HouseYear > 2020 | HouseYear < 1913')

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
2066  14556         183    4.0  151.150660   98.187482           23.0      6   
2795   2085         161    1.0   63.293079   29.389815           18.0      3   
6794  13288         161    1.0   50.808925   15.920343           20.0      8   
8828  11607         147    2.0   44.791836   28.360393            5.0      4   
9163  10814         109    1.0   37.265070   20.239714            9.0      9   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
2066         7.0       1912   0.037316         B         B        32   
2795         9.0       1912   0.014423         B         B        19   
6794         9.0       1910   0.014423         B         B        19   
8828         9.0       4968   0.319809         B         B        25   
9163        12.0   20052011   0.136330         B         B        30   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
2066      6748        93         562.0            4       10       B   
2795      3806         4           NaN            0        3       B   
6794      3806         4           NaN            0        3       B   
8828      4756        16        2857.0            5        8       B   
9163      6141        10         262.0            3        6       B   

              Price  
2066  284884.959104  
2795  262279.813915  
6794  287954.720682  
8828  243028.603096  
9163  254084.534396

Так же как и в пункте 7. Неадекатные значения 4968 и 20052011

.

8. Экология. Много 0

Сложно оценивать без графиков. Оценивать по квартилям нет навыка.

9. Social_1 Все хорошо

10. Social_2 Посмотреть сколько значений больше 10000

max сильно выше mean

In [31]:
train_df.query('(Social_2 > 14000)')

Id  DistrictId  Rooms     Square  LifeSquare  KitchenSquare  Floor  \
15     4328          96    1.0  36.673407   16.285522            9.0      3   
145    8423          96    1.0  41.888845   22.447228            8.0     10   
210    2510          27    1.0  38.737931   24.178119            8.0     14   
211    8192          27    3.0  84.274055   50.505379            8.0      7   
238     557          12    2.0  57.154361   35.821952            9.0      9   
...     ...         ...    ...        ...         ...            ...    ...   
9891   6001          96    4.0  99.263427   63.582708            9.0     11   
9893  10736          13    3.0  80.030330   47.801599           10.0     16   
9941  16031          96    1.0  35.474685   18.023801            9.0      6   
9964  12763          27    1.0  43.963812   24.734367            8.0      2   
9984  12566          13    1.0  40.234940   20.648798            7.0     10   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
15          12.0       2003   0.041116         B         B        53   
145         14.0       1995   0.041116         B         B        53   
210         14.0       2005   0.041116         B         B        53   
211          9.0       1997   0.041116         B         B        53   
238         14.0       1995   0.090799         B         B        74   
...          ...        ...        ...       ...       ...       ...   
9891        22.0       1995   0.041116         B         B        53   
9893        17.0       1992   0.090799         B         B        74   
9941        14.0       2002   0.041116         B         B        53   
9964        14.0       2006   0.041116         B         B        53   
9984        17.0       2002   0.090799         B         B        74   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
15       14892         4           NaN            1        4       B   
145      14892         4           NaN            1        4       B   
210      14892         4           NaN            1        4       B   
211      14892         4           NaN            1        4       B   
238      19083         2           NaN            5       15       B   
...        ...       ...           ...          ...      ...     ...   
9891     14892         4           NaN            1        4       B   
9893     19083         2           NaN            5       15       B   
9941     14892         4           NaN            1        4       B   
9964     14892         4           NaN            1        4       B   
9984     19083         2           NaN            5       15       B   

              Price  
15    168143.345700  
145   174939.537321  
210   182642.355518  
211   291518.545216  
238   239606.501593  
...             ...  
9891  377588.163478  
9893  311528.100863  
9941  171076.791823  
9964  187699.271984  
9984  177118.478823  

[327 rows x 20 columns]

Нет значений между 14000 и 19000 (которых 176 штук), это влияет на (max сильно выше mean)

Непонимаю что с этим делать

.

11. Social_3 Все нормально

12. Healthcare_1 всего 1 ноль и много незаполненных значений

In [32]:
train_df.loc[(train_df['Healthcare_1'].isnull())]

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
0     11809          27    3.0  115.027311         NaN           10.0      4   
3      2352           1    1.0   40.409907         NaN            1.0     10   
4     13866          94    2.0   64.285067   38.562517            9.0     16   
8      9951           1    1.0   46.887892   44.628132            1.0     12   
11    16755          88    2.0   53.975144   34.153584            8.0      2   
...     ...         ...    ...         ...         ...            ...    ...   
9993   9289           1    3.0   84.486078   88.931509            0.0      4   
9995   1260          61    2.0   49.090728   33.272626            6.0      3   
9996  16265          27    2.0   64.307684   37.038420            9.0     13   
9997   2795         178    1.0   29.648057   16.555363            5.0      3   
9999   7202          94    1.0   35.815476   22.301367            6.0      9   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
0           10.0       2014   0.075424         B         B        11   
3           22.0       1977   0.007122         B         B         1   
4           16.0       1972   0.282798         B         B        33   
8           20.0       1977   0.007122         B         B         1   
11          12.0       1978   0.127376         B         B        43   
...          ...        ...        ...       ...       ...       ...   
9993        22.0       2016   0.007122         B         B         1   
9995        12.0       1981   0.300323         B         B        52   
9996         0.0       1977   0.072158         B         B         2   
9997         5.0       1958   0.460556         B         B        20   
9999         9.0       1975   0.127376         B         B        43   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
0         3097         0           NaN            0        0       B   
3          264         0           NaN            0        1       B   
4         8667         2           NaN            0        6       B   
8          264         0           NaN            0        1       B   
11        8429         3           NaN            3        9       B   
...        ...       ...           ...          ...      ...     ...   
9993       264         0           NaN            0        1       B   
9995     10311         6           NaN            1        9       B   
9996       629         1           NaN            0        0       A   
9997      4386        14           NaN            1        5       B   
9999      8429         3           NaN            3        9       B   

              Price  
0     305018.871089  
3     168106.007630  
4     343995.102962  
8     160400.401732  
11    253124.005332  
...             ...  
9993  264213.558631  
9995  119367.455796  
9996  199715.148807  
9997  165953.912580  
9999  177685.627486  

[4798 rows x 20 columns]